dawn/src/dawn/SIR/proto/SIR/statements.proto

```diff
@@ -30,10 +31,38 @@ message SourceLocation {
   int32 Column = 2; // Column in the `Line`
 }
 
-enum LocationType {
-  Edge = 0;
-  Vertex = 1;
-  Cell = 2;
+// @brief Cartesian horizontal dimension of a field
+//
+// For a cartesian grid we enable/disable the I and J components through their respective masks.
+// @ingroup sir_proto
+message CartesianDimension {
+  int32 mask_cart_i = 1;                 // mask for cartesian component i
+  int32 mask_cart_j = 2;                 // mask for cartesian component j
+}
+
+// @brief Triangular horizontal dimension of a field
+//
+// For a triangular grid we need to specify the dense location type.
+// The dimension could be sparse. In such case the neighbor chain is set.
+// @ingroup sir_proto
+message TriangularDimension {
+  dawn.proto.enums.LocationType dense_location_type = 1; // dense location type
```

Comment:
Do we need this? The comment for neighbor_chain also suggests that we are saving redundant information in the case we have sparse dimensions.